syntax = "proto3";

option go_package = "pb/game";

import "proto/common/empty.proto";
import "proto/common/tip.proto";
import "proto/common/user_accounts.proto";

package loginpb;

message LoginSessionInfo {
  string account = 1;
  uint64 role_id = 2;
  uint64 session_id = 3;
  uint64 gate_id = 4;
  string device_id = 5;
  int64 login_time = 6;
  string fsm = 7;
}


message AccountSimplePlayerWrapper
{
  AccountSimplePlayer player = 1;
}

message LoginRequest
{
  string account = 1;
  string password = 2;
}

message LoginResponse
{
  TipInfoMessage error_message = 1;
  repeated AccountSimplePlayerWrapper players = 2;
}

message TestResponse
{
  TipInfoMessage error_message = 1;
  repeated AccountSimplePlayerWrapper players = 2;
  repeated string teststring = 3;
  repeated int32 testint = 4;
}

message CreatePlayerRequest
{
	
}

message CreatePlayerResponse
{
	TipInfoMessage error_message = 1;
	repeated AccountSimplePlayerWrapper players = 2;
}

message EnterGameRequest
{
	uint64 player_id  = 1;
}

message EnterGameResponse
{
	TipInfoMessage error_message = 1;
	uint64 player_id  = 2;
}

message LeaveGameRequest
{
}

message LoginNodeDisconnectRequest
{
  uint64 session_id = 1;
}

service ClientPlayerLogin  {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc CreatePlayer (CreatePlayerRequest) returns (CreatePlayerResponse);
  rpc EnterGame (EnterGameRequest) returns (EnterGameResponse);
  rpc LeaveGame (LeaveGameRequest) returns (Empty);
  rpc Disconnect (LoginNodeDisconnectRequest) returns (Empty);
}
